File: MathForGameDevelopers_OpenTKOpenGL31CSharp/001_PointsAndVectors/Point.cs

namespace _001_PointsAndVectors
{
    public class Point
    {
        public float X { get; set; }
        public float Y { get; set; }

        public Point(float x = 0f, float y = 0f)
        {
            X = x;
            Y = y;
        }

        public Point AddVector(Vector v)
        {
            Point p2 = new Point();
            p2.X = X + v.X;
            p2.Y = Y + v.Y;
            return p2;
        }
    }
}

File: MathForGameDevelopers_WebGLTypeScript/001_PointsAndVectors/src_client/Point.ts
import { Vector } from "./Vector";

export class Point
{
    public x: number;
    public y: number;

    public constructor(x = 0, y = 0)
    {
        this.x = x;
        this.y = y;
    }
    
    public AddVector(v: Vector): Point
    {
        let p2 = new Point();

        p2.x = this.x + v.x;
        p2.y = this.y + v.y;

        return p2;
    }
}